### PR TITLE
Add has_many/belongs_to relationship between Bet and BetOption

### PR DESCRIPTION
**Pull request by ghost**
Now every BetOption knows which bet it belongs to.

### CODE REVIEW

**Review by gregdaynes — APPROVED** [review]

File: db/migrate/20170220212242_add_bet_id_to_options.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class AddBetIdToOptions < ActiveRecord::Migration[5.0]
+  def change
+    add_column :bet_options, :bet_id, :integer
```
Comment (qr8r) [inline db/migrate/20170220212242_add_bet_id_to_options.rb:3]: Missing index

File: app/models/bet_option.rb [hunk 2]
```diff
@@ -1,6 +1,7 @@
 class BetOption < ApplicationRecord
   has_many :users, through: :user_bets
   has_many :user_bets
+  belongs_to :bet
```
Comment (qr8r) [inline app/models/bet_option.rb:4]: I am ok with this, but if you are changing the model architechture, you may want to look at the rest of the relationships involved as well.

Currently: Bet has many options through user bets, but bet option belongs to a bet directly.

**Review by qr8r — CHANGES_REQUESTED** [review]

**Comment by YarekTenko** [comment]
Forgot to add foreign key to BetOption, at this stage it will produce: "can't write unknown attribute user_bet_id" error.

File: db/migrate/20170220212242_add_bet_id_to_options.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+    add_foreign_key :bet_options, :bets
+    add_index :bet_options, :bet_id
```
Comment (isaacfreeman) [inline db/migrate/20170220212242_add_bet_id_to_options.rb:5]: 👍

**Comment by YarekTenko** [comment]
Problem: changing `has_many :options, through: :user_bets, source: :bet_option` to `has_many :options, class_name: 'BetOption'` without adding appropriate id column (I think).

File: app/models/user_bet.rb [hunk 4]
```diff
@@ -2,6 +2,7 @@ class UserBet < ApplicationRecord
   belongs_to :user
   belongs_to :bet
   belongs_to :bet_option
+  has_one    :bet_option
```
Comment (qr8r) [inline app/models/user_bet.rb]: This is wrong

File: db/migrate/20170228184631_add_foreign_key_to_bet_option.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class AddForeignKeyToBetOption < ActiveRecord::Migration[5.0]
+  def change
+    add_column :bet_options, :user_bet_id, :integer
```
Comment (qr8r) [inline db/migrate/20170228184631_add_foreign_key_to_bet_option.rb]: Why do bet options belong to user bets?

**Review by qr8r — CHANGES_REQUESTED** [review]
BetOptions should not have a user id

**Review by isaacfreeman — APPROVED** [review]

**Review by qr8r — APPROVED** [review]
Requires Rebase

**Pull request merged**